repository/assignment.go

```diff
@@ -0,0 +1,38 @@
+package repository
+
+import (
+	"github.com/GarnBarn/garnbarn-backend-go/model"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type AssignmentRepository interface {
+	CreateAssignment(assignment *model.Assignment) error
+}
+
+type assignmentRepository struct {
+	db *gorm.DB
+}
+
+func NewAssignmentRepository(db *gorm.DB) AssignmentRepository {
+	// Migrate the db
+	db.AutoMigrate(&model.Assignment{})
+
+	return &assignmentRepository{
+		db: db,
+	}
+}
+
+func (a *assignmentRepository) CreateAssignment(assignmentData *model.Assignment) error {
+	logrus.Debug("Executing Create on %T", assignmentData)
+
+	res := a.db.Create(assignmentData)
+
+	// HandleError
+	if res.Error != nil && res.Error != gorm.ErrRecordNotFound {
+		logrus.Error(res.Error)
+		return res.Error
+	}
```

Comment:
In this case, if the error happens and it is not the `gorm.ErrRecordNotFound`, It might cause the bug.